### DISCUSSION TITLE
Questions on Leaflet

### DISCUSSION THREAD

**Discussion by charlieforward9** [visgl/deck.gl#8941] [Q&A] [answered]
I used DeckGL with Google Maps about 2 years ago and am working on resuscitating this project to use more layers for more exciting visualizations.

However, since then, I started working for a company that uses Leaflet, and has existing implementations that require 10k+ elements to display at the same time. The performance of this is not great, so I was curious if:

1. [Deck.gl Leaflet](https://github.com/zakjan/deck.gl-leaflet) is still being maintained?
2. There are any existing performance comparisons of respective leaflet vs deckgl layers ?
3. Recommended approach for React Leaflet? 
     - Simply wrap the `MapContainer` in a `DeckGL` component? Or is there a more involved manner of integrating the two libraries?

**Comment by charlieforward9** [comment]
Follow up, since i just spent the whole day resuscitating it...

Whats the difference between [Leaflet.glify](https://github.com/robertleeplummerjr/Leaflet.glify/) and DeckGL in the context of Leaflet?

**Comment by Pessimistress** [comment]
We don't have an official integration module for leaflet, since there has not been enough interest from the users.
There is an officially maintained [get-started example](https://github.com/visgl/deck.gl/tree/master/examples/get-started/pure-js/leaflet).

> [Deck.gl Leaflet](https://github.com/zakjan/deck.gl-leaflet) is still being maintained?

That's a question for @zakjan

**Reply by zakjan** [reply, marked as answer]
deck.gl-leaflet is feature complete. Unless there is a breaking change in Leaflet or deck.gl, it works well.

That said, as I've replied in https://github.com/zakjan/deck.gl-leaflet/issues/12 already, combining deck.gl and Leaflet is not recommended for high-performance use cases due to overlaying two canvases. Single canvas in interleaved mode (deck.gl only or with Maplibre or another WebGL-based library) is preferred instead.

**Reply by charlieforward9** [reply]
Thank you for your responses. This information is highly valuable to me.
